### PR TITLE
Activities open in iframe editing by default [#98675138]

### PR DESCRIPTION
**Pull request by dougmartin**

### CODE REVIEW

File: app/assets/javascripts/components/search/material_header.js.coffee [hunk 1]
```diff
@@ -29,7 +29,7 @@ window.SMaterialHeaderClass = React.createClass
         (span {className: 'superTiny'},
           (SGenericLink {link: material.links.edit})
         )
-      if material.links.external_edit_iframe?
+      if material.links.external_edit_iframe? and not material.lara_activity_or_sequence
```
Comment (knowuh) [inline app/assets/javascripts/components/search/material_header.js.coffee]: I understand this code, but I am wondering what the 'reason' for it is (probably I could read the PT story?).  

I am not groking the matrix of materials with `edit`, `external_edit_iframe`, and `lara_activity_or_sequence`

File: app/assets/javascripts/components/search/material_info.js.coffee [hunk 2]
```diff
@@ -8,7 +8,10 @@ window.SMaterialInfoClass = React.createClass
 
     links = []
     links.push material.links.preview           if material.links.preview
-    links.push material.links.external_edit     if material.links.external_edit
+    if material.lara_activity_or_sequence
+      links.push material.links.external_lara_edit if material.links.external_lara_edit
```
Comment (knowuh) [inline app/assets/javascripts/components/search/material_info.js.coffee]: The matrix of edit links seems to have the following `link` types:  

* `external_lara_edit`
* `external_edit`
* `external_edit_iframe` 
* `edit`

Some of these links require checking if the material is a `lara_activity_or_sequence` for the purpose of ITSI material display.

**Comment by knowuh** [comment]
Code looks good. I am having trouble wrapping my head around the various code-paths for displaying links.  I don't have any suggestions at the moment for tidying it up, because I don't understand the various options we need to support.

**Comment by dougmartin** [comment]
The idea is that for external LARA based materials the edit button defaults to iframe editing and the edit in iframe link disappears.

**Comment by dougmartin** [comment]
The push Travis failure looks unrelated to this change, btw.

**Pull request merged**